Mongo Hotfix 1 control replica set parameter

melvinsoft opened the pull request:
We need to have a default value for replica set, in configuration this value will be `None`.

OmarIthawi commented on common/lib/xmodule/xmodule/mongo_utils.py:
```diff
@@ -33,7 +33,7 @@ def connect_to_mongodb(
     # the MongoClient class for all connections. Update/simplify this code when using
     # PyMongo 3.x.
-    if kwargs.get('replicaSet'):
+    if kwargs.pop('replicaSet', '') != '':
```
Maxi, could you please make it more noob-friendly?
Please break the code into multiple lines so it's possible to understand the lines.

As far as I understand this is a hotfix, so we need to do the following:
```suggestion
    kwargs.pop('replicaSet', '')  # Hack: Compose.io expects no replicaSet` argument
    if False:  # TODO: Hack. Revert this check to the original: `if kwargs.get('replicaSet'):
````

melvinsoft replied:
No, we need to live with this change Omar, I went for something more permanent.

OmarIthawi replied:
To be clear, I understand the urgency of this so I'll dismiss my own review so it's possible to get it merged.

That being said. I don't understand the change then as far as I know it removes `replicaSet`. Then we really need it to be noob-friendly because I cannot understand what it's supposed to do and why we chose this specific implementation.

melvinsoft replied:
@OmarIthawi I added comments in the code, but maybe is still not clear. I'm gonna reword it here and then change them again the code comments if is more clear.

In order to connect to MongoDB Atlas, we need to provide the replica set name, we were lacking that variable in our configuration playbooks, so we add it. The problem is that we need a default there, an empty string for example. The problem, is that ansible YAML is really bad handling those defaults, our only option is to handle this in the code, but an empty it ends in an error, so we need to use None. 
That's why we're changing the empty string to None. Ideally, if isn't set, we should have a way in ansible, to avoid settings the replicaSet in the `lms.auth.json` and  `cms.auth.json`.

melvinsoft commented:
@thraxil @OmarIthawi i reworked the change and added more coments, please let me know how it looks for you now. Before merge, I'm gonna squash the commit and update the commit comment.

OmarIthawi commented on common/lib/xmodule/xmodule/mongo_utils.py:
```diff
@@ -30,10 +30,18 @@ def connect_to_mongodb(
     # the MongoClient class for all connections. Update/simplify this code when using
     # PyMongo 3.x.
-    if kwargs.get('replicaSet'):
+    if kwargs.get('replicaSet') and kwargs.get('replicaSet') is not None:
```
That should be enough, `None` is falsy.

```suggestion
    if kwargs.get('replicaSet'):
```

Please do what works and let's refactor tomorrow or anytime soon.

OmarIthawi commented on common/lib/xmodule/xmodule/mongo_utils.py:
```diff
@@ -30,10 +30,18 @@ def connect_to_mongodb(
+    if kwargs.get('replicaSet', '') == '':
+        # If the replicaSet is empty let's set it as None, that is the default
+        # value that pymongo has. If not, will look for a '' replica set.
+        kwargs.get('replicaSet') = None
```
```suggestion
        kwargs.get('replicaSet') = None
```

I believe that would be a syntax error.

melvinsoft replied:
Absolutely, It's fixed now 🤦‍♂ I'm really tired already 😄

melvinsoft commented:
Closing in favor of https://github.com/appsembler/edx-configs/pull/751
